apps/learner/src/lib/states/player.svelte.ts

```diff
@@ -75,31 +80,102 @@ export class Player {
     return this.#currentTrack ? { ...this.#currentTrack } : null;
   }
 
+  /**
+   * Returns the total duration of the current track in seconds.
+   */
+  get duration() {
+    return this.#duration;
+  }
+
+  /**
+   * Returns the current playback position in seconds.
+   */
+  get progress() {
+    return this.#progress;
+  }
+
   /**
    * Plays the track with the specified metadata.
    * @param track - The track metadata.
    */
   play(track: Track) {
+    // Stop current audio if playing
+    if (this.#audio) {
+      this.#audio.pause();
+      this.#audio.currentTime = 0;
+    }
+
     this.#currentTrack = track;
+
+    this.#audio = new Audio(track.url);
```

Comment:
Are we gonna create a new `Audio` object every time we play a different audio? Can we reuse this object?